geoh5py/objects/drape_model.py

```diff
@@ -21,33 +21,54 @@
 
 import numpy as np
 
-from .grid_object import GridObject
-from .object_base import ObjectType
+from ..shared.utils import box_intersect, mask_by_extent
+from .object_base import ObjectBase, ObjectType
 
 
-class DrapeModel(GridObject):
+class DrapeModel(ObjectBase):
     """
     Drape (curtain) model object made up of layers and prisms.
+
+    :attr layers: Array of layers in the drape model organized into blocks
+        representing each prism in the model.
+    :attr prisms: Array detailing the assembly of
+        :obj:`geoh5py.objects.drape_model.DrapeModel.layers` within the trace
+         of the drape model.
     """
 
     __TYPE_UID = uuid.UUID("{C94968EA-CF7D-11EB-B8BC-0242AC130003}")
-
-    def __init__(self, object_type: ObjectType, **kwargs):
-        self._layers: np.ndarray | None = None
-        self._prisms: np.ndarray | None = None
-
-        super().__init__(object_type, **kwargs)
+    __LAYERS_DTYPE = np.dtype([("I", "<i4"), ("K", "<i4"), ("Bottom elevation", "<f8")])
+    __PRISM_DTYPE = np.dtype(
+        [
+            ("Top easting", "<f8"),
+            ("Top northing", "<f8"),
+            ("Top elevation", "<f8"),
+            ("First layer", "<i4"),
+            ("Layer count", "<i4"),
+        ]
+    )
+
+    def __init__(
+        self,
+        object_type: ObjectType,
+        layers: np.ndarray | list | tuple | None = None,
+        prisms: np.ndarray | list | tuple | None = None,
+        **kwargs,
+    ):
+        self._centroids: np.ndarray | None = None
+        self._layers: np.ndarray = self.validate_layers(layers)
+        self._prisms: np.ndarray = self.validate_prisms(prisms)
+
+        super().__init__(object_type, layers=layers, prisms=prisms, **kwargs)
```

Comment:
Ben missed this one ;-) 

ObjectBase doens't have layers nor prisms

Reply:
Done